securedrop/tests/test_manage.py

```diff
@@ -52,3 +59,128 @@ def test_exception_handling_when_duplicate_username(self, mock_raw_input,
         self.assertEqual(return_value, 1)
         self.assertIn('ERROR: That username is already taken!',
                       sys.stdout.getvalue())
+
+
+class TestManage(object):
+
+    def setup(self):
+        utils.env.setup()
+
+    def teardown(self):
+        utils.env.teardown()
+
+    def test_translate_compile_code_and_template(self):
+        source = [
+            'tests/i18n/code.py',
+            'tests/i18n/template.html',
+        ]
+        kwargs = {
+            'translations_dir': config.TEMP_DIR,
+            'mapping': 'tests/i18n/babel.cfg',
+            'source': source,
+            'extract_update': True,
+            'compile': True,
+            'verbose': logging.DEBUG,
+        }
+        args = argparse.Namespace(**kwargs)
+        manage.setup_verbosity(args)
+        manage.translate(args)
+        messages_file = os.path.join(config.TEMP_DIR, 'messages.pot')
+        assert os.path.exists(messages_file)
+        pot = open(messages_file).read()
+        assert 'code hello i18n' in pot
+        assert 'template hello i18n' in pot
+
+        locale = 'en_US'
+        locale_dir = os.path.join(config.TEMP_DIR, locale)
+        manage.sh("pybabel init -i {} -d {} -l {}".format(
+            messages_file,
+            config.TEMP_DIR,
+            locale,
+        ))
+        mo_file = os.path.join(locale_dir, 'LC_MESSAGES/messages.mo')
+        assert not os.path.exists(mo_file)
+        manage.translate(args)
+        assert os.path.exists(mo_file)
+        mo = open(mo_file).read()
+        assert 'code hello i18n' in mo
+        assert 'template hello i18n' in mo
+
+    def test_translate_compile_arg(self):
+        source = [
+            'tests/i18n/code.py',
+        ]
+        kwargs = {
+            'translations_dir': config.TEMP_DIR,
+            'mapping': 'tests/i18n/babel.cfg',
+            'source': source,
+            'extract_update': True,
+            'compile': False,
+            'verbose': logging.DEBUG,
+        }
+        args = argparse.Namespace(**kwargs)
+        manage.setup_verbosity(args)
+        manage.translate(args)
+        messages_file = os.path.join(config.TEMP_DIR, 'messages.pot')
+        assert os.path.exists(messages_file)
+        pot = open(messages_file).read()
+        assert 'code hello i18n' in pot
+
+        locale = 'en_US'
+        locale_dir = os.path.join(config.TEMP_DIR, locale)
+        manage.sh("pybabel init -i {} -d {} -l {}".format(
+            messages_file,
+            config.TEMP_DIR,
+            locale,
+        ))
+        mo_file = os.path.join(locale_dir, 'LC_MESSAGES/messages.mo')
+
+        #
+        # Extract+update but do not compile
+        #
+        assert not os.path.exists(mo_file)
+        manage.translate(args)
+        assert not os.path.exists(mo_file)
+
+        #
```

Comment:
Can we split this into two separate tests? `test_translate_extract_and_update_without_compile` and `test_translate_compile_without_extract_and_update`?